src/DGtal/dec/DiscreteExteriorCalculus.h

```diff
@@ -101,35 +102,39 @@ namespace DGtal
     BOOST_STATIC_ASSERT(( boost::is_same<Dimension, Order>::value ));
 
     /**
-     * Static dimension.
+     * Static dimensions.
      */
-    BOOST_STATIC_CONSTANT( Dimension, dimension = dim );
+    BOOST_STATIC_ASSERT(( dim_ambient >=  dim_embedded ));
 
-    typedef DGtal::KhalimskySpaceND<dim, TInteger> KSpace;
+    BOOST_STATIC_CONSTANT( Dimension, dimension_embedded = dim_embedded );
+    BOOST_STATIC_CONSTANT( Dimension, dimension_ambient = dim_ambient );
+
+    typedef DGtal::KhalimskySpaceND<dim_ambient, TInteger> KSpace;
+    typedef typename KSpace::Cell Cell;
     typedef typename KSpace::SCell SCell;
     typedef typename KSpace::Point Point;
 
     /**
      * Cells data stuct.
-     * Holds size ratio, indexes and display_flipped for each cell of the structure.
+     * Holds size ratio, indexes and flipped for each cell of the structure.
```

Comment:
since these properties are crucial for developers, could you please add a short description of each quantity ?